runner/runners/status_manager.go

```diff
@@ -130,6 +130,12 @@ func (s *StatusManager) Query(q runner.Query, wait runner.Wait) ([]runner.RunSta
 		return []runner.RunStatus{st}, s.svcStatus, nil
 	case <-timeout:
 		return nil, s.svcStatus, nil
+	case <-wait.AbortCh:
+		// we just consumed the abort request without really aborting the run,
+		// put a new request on the channel so taskRunner will abort the run.
+		wait.AbortCh <- 1
```

Comment:
done - and agree its klunky and dangerous- we have to make sure that all Query implementations put the abort back on the channel